torch_geometric/nn/pool/glob.py

```diff
@@ -25,19 +29,19 @@ def global_add_pool(x: Tensor, batch: Optional[Tensor],
     """
     if batch is None:
         return x.sum(dim=-2, keepdim=x.dim() == 2)
-    size = int(batch.max().item() + 1) if size is None else size
-    return scatter(x, batch, dim=-2, dim_size=size, reduce='add')
+    return sum_aggr(x, batch, dim_size=size)
```

Comment:
Can we leave the implementation as it is? I am not super happy with having global modules here.